src/Utilities/Compiler/Extensions/IOperationExtensions.cs

```diff
@@ -508,12 +508,27 @@ void ProcessLocalOrParameter(ISymbol symbol)
 
         private static readonly ImmutableArray<OperationKind> s_LambdaAndLocalFunctionKinds =
             ImmutableArray.Create(OperationKind.AnonymousFunction, OperationKind.LocalFunction);
+
         public static bool IsWithinLambdaOrLocalFunction(this IOperation operation, [NotNullWhen(true)] out IOperation? containingLambdaOrLocalFunctionOperation)
         {
             containingLambdaOrLocalFunctionOperation = operation.GetAncestor(s_LambdaAndLocalFunctionKinds);
             return containingLambdaOrLocalFunctionOperation != null;
         }
 
+        public static bool IsWithinExpressionTree(this IOperation operation,
+            [NotNullWhen(true)] INamedTypeSymbol? linqExpressionTreeType)
```

Comment:
I have seen two different approaches in the codebase and I am not sure which one is the preferred one:
1. Take the named type
2. Take the compilation and retrieve the named type from within the helper